### PR TITLE
feat(blog): Migrating From WordPress to Gatsby

### PR DESCRIPTION
**Pull request by taniarascia**
## Description

Cross post of [Migrating From WordPress to Gatsby](https://www.taniarascia.com/migrating-from-wordpress-to-gatsby/).

### CODE REVIEW

File: docs/blog/2019-03-19-migrating-from-wordpress-to-gatsby/index.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+I've had quite a bit of experience with WordPress. Once I learned how to create a WordPress blog, I wrote [Developing a WordPress Theme from Scratch](https://www.taniarascia.com/developing-a-wordpress-theme-from-scratch/), a post that has had millions of hits, hundreds of comments, made me plenty of friends, and launched my blogging career. I also worked as a WordPress developer for two years. For the past four years, the blog had been running on WordPress. It was as fast, as custom, and as free of plugins as I could possibly make it, but WordPress is cumbersome, and there's only so fast and pleasant to use you can make it.
+
+After 10 days of obsessively working non-stop, I've finally migrated my site over to a [Gatsby](/)! You may notice the site is a little faster now:
+
+![](./speed.png)
```
Comment (marcysutton) [inline docs/blog/2019-03-19-migrating-from-wordpress-to-gatsby/index.md]: Can you add some alt text to the image that describes what's going on?

File: docs/blog/2019-03-19-migrating-from-wordpress-to-gatsby/index.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+Since I primarily write JavaScript these days, I wanted an SSG that runs on Node.js, and if it uses React, even better. I tested out a few sites that run on Gatsby and yeah - they were fast. Blazingly fast.
+
+### A few things I really like about Gatsby
+
+- **No page reloads** - this site is now a SPA (single page app), and clicking on any internal page from within the website doesn't need to load a completely new resource
+- **Image optimization** - all the images are automatically stripped of metadata, optimized, resized, lazy-loaded, and compressed
+- **Pre-fetch resources** - Gatsby detects what links are available on a given page and loads that data into the cache
+- **Bundling and minification** - code is minified, bundled, and served
+- **No server side rendering**
```
Comment (marcysutton) [inline docs/blog/2019-03-19-migrating-from-wordpress-to-gatsby/index.md]: Gatsby does server rendering though...?

File: docs/blog/2019-03-19-migrating-from-wordpress-to-gatsby/index.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+## My Migration Process
+
+I've been putting off migrating to a static site for months and months, because I knew it would be a huge time sink, obsession, and a lot of work. I had over 100 guides and tutorials to migrate, and in the end I was able to move everything in 10 days, so it was far from the end of the world.
+
+If you've been thinking about moving your blog from WordPress to a static site but have been putting it off due to fear of how long it will take and how much work it will be, I highly recommend giving it a shot. I'll give you the basics of what I did in case you also want to make the switch.
+
+- First, I downloaded the XML from WordPress in the **Tools -> Export** section.
+- I used the [ExitWP](https://github.com/thomasf/exitwp) tool to convert the XML to Markdown. This did about 50% of the work of converting the posts.
+- I converted tables to markdwon with the [HTML to Markdown Table Converter](https://jmalarcon.github.io/markdowntables/).
```
Comment (marcysutton) [inline docs/blog/2019-03-19-migrating-from-wordpress-to-gatsby/index.md]:
```suggestion
- I converted tables to markdown with the [HTML to Markdown Table Converter](https://jmalarcon.github.io/markdowntables/).
```

File: docs/blog/2019-03-19-migrating-from-wordpress-to-gatsby/index.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+You can [view the source of the completed website](https://github.com/taniarascia/taniarascia.com).
+
+From there it was just a matter of building out all the pages, learning enough GraphQL to make the proper queries, and doing a lot of small cleanup and tweaks.
+
+## Conclusion
+
+It took a bit of work, but the [taniarascia.com](https://www.taniarascia.com) blog is now built with React, Node.js and Gatsby, and hosted on Netlify. I feel great that all of my posts are now safely saved in version control and markdown. It's a relief for me to know that they're no longer an HTML mess inside of a MySQL database, but markdown files which are easy to read, write, edit, share, and backup.
+
+There is a good amount of prerequisite knowlegde required to set up a Gatsby site - HTML, CSS, JavaScript, ES6, Node.js development environment, React, and GraphQL are the major ones. If you're a complete beginner to all of these technologies, setting up a Gatsby blog might be a bit of a challenge. However, the [Gatsby Getting Started Tutorial](/tutorial/) is a complete step-by-step guide that covers most of these topics, and is a great choice for an intermediate developer.
```
Comment (marcysutton) [inline docs/blog/2019-03-19-migrating-from-wordpress-to-gatsby/index.md]:
```suggestion
There is a good amount of prerequisite knowledge required to set up a Gatsby site - HTML, CSS, JavaScript, ES6, Node.js development environment, React, and GraphQL are the major ones. If you're a complete beginner to all of these technologies, setting up a Gatsby blog might be a bit of a challenge. However, the [Gatsby Getting Started Tutorial](/tutorial/) is a complete step-by-step guide that covers most of these topics, and is a great starting point.
```
We should encourage all developers to give it a shot, at least!

File: docs/blog/2019-03-19-migrating-from-wordpress-to-gatsby/index.md (hunk 2 above)
Comment (amberleyromo) [reply docs/blog/2019-03-19-migrating-from-wordpress-to-gatsby/index.md]: It prerenders to build and deliver static assets. @DSchau 	how do you normally like to make this distinction.

Comment (DSchau) [reply docs/blog/2019-03-19-migrating-from-wordpress-to-gatsby/index.md]: Gatsby does server-side rendering at build time.

It doesn't _require_ a Node.js server (e.g. like Next.js does), which saves a lot of 🤑.

```suggestion
- **Server-side rendered, at build time** Gatsby builds optimized static assets, which can be hosted anywhere!
```

**Comment by taniarascia** [comment]
Thanks for the feedback! I'll make those changes and update the PR today.

**Comment by taniarascia** [comment]
Updates all set.

**Pull request merged**